WiseBuy/View Controllers/FindConnectionsViewController.m

```diff
@@ -76,53 +76,56 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     FriendCell *cell = [self.tableView dequeueReusableCellWithIdentifier:kFriendCellIdentifier];
-    PFUser *friend = self.friends[indexPath.row];
-    cell.cellUser = friend;
-    cell.nameLabel.text = [NSString stringWithFormat:@"%@%@", friend[kFirstName] , friend[kLastName]];
-    cell.usernameLabel.text = friend[kUsername];
-    
-    PFFileObject *userImage = cell.cellUser[kImage];
-    NSURL *url = [NSURL URLWithString:userImage.url];
-    NSData *data = [NSData dataWithContentsOfURL:url];
-    UIImage *img = [[UIImage alloc] initWithData:data];
-    cell.profileImageView.image = img;
-    
-    if(self.segControl.selectedSegmentIndex == 1) {
-        [cell.addFriendButton setTitle:kRemove forState:UIControlStateNormal];
-        [cell.addFriendButton setTintColor:[UIColor systemRedColor]];
-        cell.chosenMode = 1;
-    } else if (self.segControl.selectedSegmentIndex == 0) {
-        [cell.addFriendButton setTitle:kAdd forState:UIControlStateNormal];
-        cell.chosenMode = 0;
-        if ([self.user[kOutgoingFriendRequests] containsObject:friend.username]) {
-            cell.addFriendButton.tintColor = [UIColor systemTealColor];
-            [cell.addFriendButton setTitle:kCancel forState:UIControlStateNormal];
+    if (self.friends > 0) {
```

Comment:
Good job checking for error conditions. If you want to be thorough, you also need to check that indexPath.row doesn't exceed the upper bound of the array as well.

Having said that, I personally don't think this needs to be super thorough because if the table row index gets out of sync with the data source, something is catastrophically wrong and the user would probably need to restart the app anyway. This is likely a less common error case compared to things like handling edge cases when parsing data from an external API.